torch/csrc/jit/ir.cpp

```diff
@@ -853,10 +854,54 @@ Value* Node::namedInput(Symbol name) const {
   return input(findArgument(schema(), name));
 }
 
+bool Node::matches(const FunctionSchema& schema) const {
+  // wrong name
+  if (kind().toQualString() != schema.name()) {
+    return false;
+  }
+  at::ArrayRef<const Value*> actuals = inputs();
+  const auto& formals = schema.arguments();
+
+  // not enough inputs
+  if (actuals.size() < formals.size()) {
+    return false;
+  }
+
+    TypeEnv type_env;
+  for (size_t i = 0; i < formals.size(); ++i) {
+    auto formal = formals[i].type();
+    const MatchTypeReturn matched_type = matchTypeVariables(
+        formal, actuals[i]->type(), type_env);
+    if (!matched_type.success()) {
+      return false;
+    }
+
+    TypePtr resolved = tryEvalTypeVariables(formal, type_env);
+    if (resolved) {
+      formal = resolved;
+    }
+    // note: it is possible at this point that type variable matching has
+    // not resolved all type variables, e.g. if None was matched to Optional[T]
+    // we will not succeed at matching T. However None <: Optional[T] so this
+    // check can still succeed.
+
+    if (!actuals[i]->type()->isSubtypeOf(formal)) {
+      return false;
+    }
+  }
+
+  // too many inputs
+  if (!schema.is_vararg() && actuals.size() != formals.size()) {
+    return false;
+  }
+
+  return true;
+}
+
 bool Node::matches(
```

Comment:
I know this function isn't part of this work, but is it worth adding a comment header here? Schema matching is obvious enough but it's not immediately clear why we're doing this other locally-unrelated thing on `const_inputs` at the same time (also! `should_be_const_inputs` per this afternoon :P)

No need for diff creep but thinking out loud as I read through this